src/ray/core_worker/core_worker.cc

```diff
@@ -4327,6 +4327,20 @@ void CoreWorker::RecordTaskLogEnd(int64_t stdout_end_offset,
       worker::TaskStatusEvent::TaskStateUpdate(task_log_info));
 }
 
+void CoreWorker::UpdateTaskIsDebuggerPaused(const bool is_debugger_paused) {
+  auto current_task = worker_context_.GetCurrentTask();
+  RAY_CHECK(current_task)
+      << "We should have set the current task spec before executing the task.";
+  RAY_LOG(DEBUG) << "Task " << current_task->TaskId() << " is paused by debugger set to"
+                 << is_debugger_paused;
+  task_manager_->RecordTaskStatusEvent(
+      current_task->AttemptNumber(),
+      *current_task,
```

Comment:
I think this may not work with async actor (it is due to nasty tech debt in ray core). There's a hacky workaround we can use, and we may need to use the same workaround for this.

Can you try adding unit tests and see if it works with async actor / threaded actor / regular actor? 

- async actor: actor with async method
- threaded actor: actor with num_concurrency > 1 inside ray.remote
- regular actor: the actor you know

If it turns out async actor doesn't work (which I suspect), let me give you a code pointer here.

Reply:
Can you point me to unit tests that test similar logic. I looked at core_worker_test but didn't find anything relevant.